### PR TITLE
Add test harness for llbc pretty output

### PR DESCRIPTION
**Pull request by Nadrieril**
This will make it a lot easier to debug issues like https://github.com/AeneasVerif/aeneas/issues/97

### CODE REVIEW

**Comment by Nadrieril** [comment]
I managed to break CI!

**Comment by Nadrieril** [comment]
Converting to draft while I improve it

**Comment by Nadrieril** [comment]
It's ready!

File: charon/src/cli_options.rs
Comment (sonmarcho) [inline charon/src/cli_options.rs:135]: It would be better to name this `no-serialize` as we can serialize ULLBC as well?

File: charon/src/charon-driver.rs
Comment (sonmarcho) [inline charon/src/charon-driver.rs:236]: It is good to make the `CharonFailure` kind explicit here (useful when we update this enumeration, to make sure we don't forget any case)

File: charon/tests/ui.rs
Comment (sonmarcho) [inline charon/tests/ui.rs:6]: I think there is a different syntax to refer to a definition in a comment? Something like [HELP_STRING]?...

File: charon/tests/ui.rs
Comment (sonmarcho) [inline charon/tests/ui.rs:29]: Why "magic"?

File: charon/tests/ui.rs
Comment (sonmarcho) [inline charon/tests/ui.rs:51]: Why not return after we successfully parse the first magic comment?

File: charon/tests/ui/issue-92-nonpositive-variant-indices.rs
Comment (sonmarcho) [inline charon/tests/ui/issue-92-nonpositive-variant-indices.rs:1]: My understanding is that this PR requires https://github.com/AeneasVerif/charon/pull/107 to be merged before?

File: charon/tests/ui/issue-97-missing-parent-item-clause.rs
Comment (sonmarcho) [inline charon/tests/ui/issue-97-missing-parent-item-clause.rs:1]: Same: we need issue 97 to be merged before the current PR?

File: charon/tests/ui/issue-92-nonpositive-variant-indices.rs
Comment (Nadrieril) [reply charon/tests/ui/issue-92-nonpositive-variant-indices.rs:1]: No, I just added a few test cases from open issues. As you can see, CI passes (as you can see, the match is weird)

Comment (sonmarcho) [reply charon/tests/ui/issue-92-nonpositive-variant-indices.rs:1]: Indeed

File: charon/tests/ui.rs
Comment (Nadrieril) [reply charon/tests/ui.rs:29]: I think this is what the ui-test crate calls them. I could call them something else if you have an idea

File: charon/tests/ui.rs
Comment (Nadrieril) [reply charon/tests/ui.rs:51]: Yeah fair the loop doesn't make sense yet. I wrote it like this because I plan to add other options, e.g. `//@ driver-opts=--no-code-duplication` depending on what comes useful.

File: charon/src/cli_options.rs
Comment (sonmarcho) [inline charon/src/cli_options.rs:131]: Update this message: it can be the ULLBC

File: charon/tests/ui.rs
Comment (sonmarcho) [reply charon/tests/ui.rs:50]: Makes sense, and it's not that important anyway

**Review by sonmarcho — APPROVED** [review]
Looks good to me! I just have one comment I want to see you address: once it's done, you can directly merge the PR (no need to wait for me to reapprove and merge).

**Pull request merged**
